tap_linear/client.py

```
@@ -77,21 +79,54 @@ def parse_response(self, response: requests.Response) -> Iterable[dict]:
         resp_json = response.json()
         for row in resp_json["data"][self.name]["nodes"]:
             yield row
-    
-    def validate_response(self, response: requests.Response) -> None:
 
+    def get_response_status_code(self, response: requests.Response) -> int:
+        """Try to get the status_code from the body if it's an error response.
+            For example, if the response is a 429 error, the response.status_code will be 400.
+        """
+        status_code = response.status_code
+        try:
+            if status_code >= 400:
+                for error in response.json().get("errors", []):
+                    code = error.get("extensions", {}).get("statusCode")
+                    if code:
+                        status_code = code
+                        break
+        except Exception as e:
+            self.logger.error(f"Error parsing response status code: {response.text} {e}")
+
+        return status_code
+
+    def handle_rate_limit_error(self, response: requests.Response) -> None:
+        """Handle rate limit error."""
+        self.logger.error(f"Rate limit exceeded: {response.text}")
+        retry_after_timestamp = response.headers.get("x-ratelimit-requests-reset")
+        if retry_after_timestamp:
+            retry_after = int(retry_after_timestamp) / 1000
+            now = datetime.datetime.now(datetime.timezone.utc)
+            dt = datetime.datetime.fromtimestamp(retry_after, tz=datetime.timezone.utc)
+            retry_after = (dt - now).total_seconds()
+            self.logger.info(f"Rate limit exceeded. Retrying in {retry_after} seconds.")
+            time.sleep(retry_after)
+        else:
+            self.logger.error(f"Rate limit exceeded. No retry after timestamp found.")
```

so if there's no "x-ratelimit-requests-reset" does it mean we can't make more requests for the day? if so we probably should raise a fatal error instead of retrying

---

the request limit is 1500 req/hour. So it should be less than a day, in my tests it was one hour. I can add a condition to raise a fatal error if retry_after is greater than X amount, what do you think X should be?

---

I see, if there's no daily limit I think this should be fine